testbench/grpc_server.py

```diff
@@ -56,6 +57,39 @@ def CreateBucket(self, request, context):
         self.db.insert_bucket(bucket, context)
         return bucket.metadata
 
+    def ListBuckets(self, request, context):
+        if not request.parent.startswith("projects/"):
+            return testbench.error.invalid(
+                "invalid format for parent=%s" % request.parent, context
+            )
+        project = request.parent[len("projects/") :]
+        if len(request.read_mask.paths) == 0:
+            # By default we need to filter out `acl`, `default_object_acl`, and `owner`
+            def filter(bucket):
+                b = storage_pb2.Bucket()
+                b.CopyFrom(bucket)
+                b.ClearField("acl")
+                b.ClearField("default_object_acl")
+                b.ClearField("owner")
+                return b
+
+        elif request.read_mask.paths == ["*"]:
```

Comment:
I guess this is the equivalent of `projection=full` in JSON?

Reply:
I think so, I am going by the comments in the proto file.